docs/Testing.md

```diff
@@ -0,0 +1,69 @@
+# Testing
+
+Moya has been created with testing at its heart. In this document, you will find all the customization points that will allow you to tailor your testing.
+
+## `sampleData`
+
+When creating your `TargetType` you are required to provide `sampleData` for your targets. All you need to do there is to provide `Data` that represents a sample response from every particular target.
+
+For example:
+
+```swift
+public var sampleData: Data {
+    switch self {
+    case .userRepositories(let name):
+        return "[{\"name\": \"Repo Name\"}]".data(using: String.Encoding.utf8)!
+    }
+}
+```
+
+This can be used later for tests or just for providing offline support while developing.
+
+## `stubClosure`
+
```

Comment:
I feel like this section is a little unclear but we could expand on this later in another PR?  I would like it to feel a bit more explanatory.


> The `stubClosure` property of a `MoyaProvider` is responsible for configuring whether a provider returns an actual or stubbed response. The `stubClosure` is responsible for mapping a `TargetType` to a case of the `StubBehavior` enum:

Then I would list all cases of `StubBehavior` and explain what each of them does.

I just think it's clearer this way and also matches the setup of the next section. 😓

Reply:
Yeah I like it better as well.